docs/proposals/scheduling/multi-scheduling-group/README.md

```diff
@@ -0,0 +1,366 @@
+
+### Goals
+
+- Extend the API of PropagationPolicy to hold multiple affinity groups declaration.
+- Extend the API of ResourceBinding to annotate the affinity group that the scheduler is inspecting.
+- Propose the implementation ideas for involved components, including `karmada-controller-manager`, `karmada-webhook` and `karmada-scheduler`.
+
+### Non-Goals
+
+- **The relative priority of clusters in the same group**
+
+This proposal focuses on introducing multiple affinity groups, for the relative
+priority of clusters in the same group topic is out of scope. And such requirements
+could be done by `weightPreference`(.spec.placement.replicaScheduling.weightPreference) or
+define a strategy for `karmada-scheduler` to score specified clusters.
+
+- **Scheduling re-balance**
+
+Indeed, oftentimes people want the `karmada-scheduler` to perform an extra schedule
+for different purposes, like the question addressed by the discussion
+[#3069](https://github.com/karmada-io/karmada/discussions/3069), these kind of
+issues should be tracked by another proposal.
+
+## Proposal
+
+### User Stories (Optional)
+
+#### As a user, I prefer to deploy my applications on clusters in local data center to save costs.
+
+I have some clusters in my data center as well as some managed clusters from cloud
+providers(like AWS and Google Cloud). Since the cost of the managed clusters is higher
+than private clusters, so I prefer to deploy applications on private clusters and
+take the managed clusters as a backup.
+
+#### As a user, I want to deploy applications on the primary cluster and leave a backup cluster for disaster recovery cases.
+
+I have two clusters, the primary, and the backup cluster, I want Karmada to deploy
+my applications on the primary cluster and migrate applications on the backup cluster
+when the primary cluster becomes unavailable due to like data center power off or
+maintenance task.
+
+### Notes/Constraints/Caveats (Optional)
+
+This proposal mainly focuses on the changes in `PropagationPolicy` and `ResourceBinding`,
+but it will also be applied to `ClusterPropagationPolicy` and `ClusterResourceBinding`.
+
+### Risks and Mitigations
+
+This proposal maintains the backward compatibility, the system built with previous
+versions of Karmada can be seamlessly migrated to the new version.
+The previous configurations(yamls) could be applied to the new version of Karmada and
+without any behavior change.
+
+## Design Details
+
+### API change
+
+#### PropagationPolicy API change
+
+This proposal proposes a new field `OrderedClusterAffinities` for declaring
+multiple affinity terms in `.spec.placement` of `PropagationPolicy`.
+```yaml
+// Placement represents the rule for select clusters.
+type Placement struct {
+	// ClusterAffinity represents scheduling restrictions to a certain set of clusters.
+	// If not set, any cluster can be scheduling candidate.
+	// +optional
+	ClusterAffinity *ClusterAffinity `json:"clusterAffinity,omitempty"`
```

Comment:
The newly added `OrderedClusterAffinities` field has overwritten the function of this field(When the length of the new field is 1), will this field be deprecated?